pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -231,8 +232,12 @@ func (r *Reconciler) getProviderSecretData(ctx context.Context, providerClient p
 }
 
 func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// prevent reconcile on status updates
+	// https://github.com/kubernetes-sigs/kubebuilder/issues/618#issuecomment-698018831
+	pred := predicate.GenerationChangedPredicate{}
```

Comment:
I mentioned this in a previous comment https://github.com/external-secrets/external-secrets/pull/110#discussion_r611946630, but because this is is the controller for the resource which manages the status, this controller should reconcile on status changes.

Reply:
Yes, that makes sense! I ran into an issue where the `condition[].message` changed with every reconciliation (the error contained a aws `requestID`).